packages/bigtable/src/cluster.js

```diff
@@ -0,0 +1,247 @@
+/*!
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module bigtable/instance
+ */
+
+'use strict';
+
+var common = require('@google-cloud/common');
+var format = require('string-format-obj');
+var is = require('is');
+var nodeutil = require('util');
+
+/**
+ * Create a cluster object to interact with your cluster.
+ *
+ * @constructor
+ * @alias module:bigtable/cluster
+ *
+ * @param {string} name - Name of the cluster.
+ *
+ * @example
+ * var instance = bigtable.instance('my-instance');
+ * var cluster = instance.cluster('my-cluster');
+ */
+function Cluster(instance, name) {
+  var id = instance.id + '/clusters/' + name;
+
+  var methods = {
+
+    /**
+     * Create a cluster.
+     *
+     * @param {object} options - See {module:bigtable/instance#createCluster}
+     *
+     * @example
+     * cluster.create(function(err, cluster, operation, apiResponse) {
+     *   if (err) {
+     *     // Error handling omitted.
+     *   }
+     *
+     *   operation
+     *     .on('error', console.error)
+     *     .on('complete', function() {
+     *       // The cluster was created successfully.
+     *     });
+     * });
+     */
+    create: true,
+
+    /**
+     * Delete the cluster.
+     *
+     * @param {function=} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * cluster.delete(function(err, apiResponse) {});
+     */
+    delete: {
+      protoOpts: {
+        service: 'BigtableInstanceAdmin',
+        method: 'deleteCluster'
+      },
+      reqOpts: {
+        name: id
+      }
+    },
+
+    /**
+     * Check if a cluster exists.
+     *
+     * @param {function} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {boolean} callback.exists - Whether the cluster exists or not.
+     *
+     * @example
+     * cluster.exists(function(err, exists) {});
+     */
+    exists: true,
+
+    /**
+     * Get a cluster if it exists.
+     *
+     * @example
+     * cluster.get(function(err, cluster, apiResponse) {
+     *   // The `cluster` data has been populated.
+     * });
+     */
+    get: true,
```

Comment:
Technically yes, however create returns an operation, so I'm not really sure how that is supposed to work? Also it turns out that upstream CreateCluster rpc is not fully functional yet.